### PR TITLE
test: pipeline_buffer_connect tests added

### PR DESCRIPTION
**Pull request by dabekjakub**
Added tests for pipeline_buffer_connect function.

Signed-off-by: Jakub Dabek <jakub.dabek@linux.intel.com>

### CODE REVIEW

File: test/cmocka/src/audio/pipeline/pipeline_buffer_connect.c [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+#include <sof/audio/component.h>
+#include <sof/audio/pipeline.h>
+#include <sof/schedule.h>
+#include "pipeline_mocks.h"
+#include "pipeline_connection_mocks.h"
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+	static int setup(void **state)
```
Comment (jajanusz) [inline test/cmocka/src/audio/pipeline/pipeline_buffer_connect.c]: Please fix formatting

File: test/cmocka/src/audio/pipeline/pipeline_buffer_connect.c [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+{
+	*state = get_standard_connect_objects();
+	return 0;
+}
```
Comment (jajanusz) [inline test/cmocka/src/audio/pipeline/pipeline_buffer_connect.c]: Need newline

**Comment by FocusLuo** [comment]
Can one of the admins verify this patch?

**Comment by jajanusz** [comment]
@dabekjakub 
Please update or close if you are not going to update this PR.

**Pull request closed without merging**